2-Authorization-I/1-call-graph/SPA/src/app/account-switch/account-switch.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import {
+  MsalService,
+  MSAL_GUARD_CONFIG,
+  MsalGuardConfiguration,
+} from '@azure/msal-angular';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+import {
+  RedirectRequest,
+  InteractionRequiredAuthError,
+  SilentRequest,
+  AccountInfo,
+} from '@azure/msal-browser';
+
+@Component({
+  selector: 'app-account-switch-component',
+  templateUrl: './account-switch.component.html',
+  styleUrls: ['./account-switch.component.css'],
+})
+export class AccountSwitchComponent implements OnInit {
+  accounts: AccountInfo[] = [];
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    @Inject(MSAL_GUARD_CONFIG) private msalGuardConfig: MsalGuardConfiguration,
+    private authService: MsalService
+  ) {}
+
+  ngOnInit(): void {}
+
+  switchAccount(account: AccountInfo | null) {
+    const activeAccount = this.authService.instance.getActiveAccount();
+    if (!account) {
```

Comment:
```suggestion
    if (!account) {
```

so if `account` is falsy, how are we setting the active account with `account`?

Reply:
If the account is falsy you sent the active account to null. check the code below:
```html
<li *ngFor="let account of data.accounts" mat-dialog-close (click)="switchAccount(account)" mat-dialog-close="true">
            <mat-icon>account_circle</mat-icon>
            <span >{{account.username}}</span>
        </li>
       <li (click)="switchAccount(null)">
         <mat-icon>library_add</mat-icon>
         <span>New account</span>
       </li>
```
If the switchAccount is called with the value null. we sent the active account to null first then login in the user to add a new account.